Run macOS jobs on Azure Pipelines using macOS 13

gsnedders opened the pull request:
While this doesn't appear to have been announced anywhere, it has once again silently launched at the same time as GitHub Actions gained availability. There seems to be little downside to just doing this now, so let's do it.

https://dev.azure.com/web-platform-tests/wpt/_build/results?buildId=99408&view=results is a STP run, https://dev.azure.com/web-platform-tests/wpt/_build/results?buildId=99411&view=results is a Safari (stable) run.

(These differ only in commit message, not in content.)

gsnedders committed:
Run macOS jobs on Azure Pipelines using macOS 13

While this doesn't appear to have been announced anywhere, it has once
again silently launched at the same time as GitHub Actions gained
availability. There seems to be little downside to just doing this
now, so let's do it.

gsnedders commented:
wpt.fyi results: [Safari stable changes](https://wpt.fyi/results/?q=seq%28status%3Apass%7Cstatus%3Aok%20status%3Apass%7Cstatus%3Aok%20status%3Apass%7Cstatus%3Aok%20status%3Apass%7Cstatus%3Aok%20status%3A%21pass%26status%3A%21ok%26status%3A%21missing%29%20or%20seq%28status%3A%21pass%26status%3A%21ok%26status%3A%21missing%20status%3A%21pass%26status%3A%21ok%26status%3A%21missing%20status%3A%21pass%26status%3A%21ok%26status%3A%21missing%20status%3A%21pass%26status%3A%21ok%26status%3A%21missing%20status%3Apass%7Cstatus%3Aok%29&run_id=5110080628850688&run_id=6256187551514624&run_id=5160666317455360&run_id=5161652347994112&run_id=6300854842294272), [STP changes](https://wpt.fyi/results/?q=seq%28status%3Apass%7Cstatus%3Aok%20status%3Apass%7Cstatus%3Aok%20status%3Apass%7Cstatus%3Aok%20status%3Apass%7Cstatus%3Aok%20status%3A%21pass%26status%3A%21ok%26status%3A%21missing%29%20or%20seq%28status%3A%21pass%26status%3A%21ok%26status%3A%21missing%20status%3A%21pass%26status%3A%21ok%26status%3A%21missing%20status%3A%21pass%26status%3A%21ok%26status%3A%21missing%20status%3A%21pass%26status%3A%21ok%26status%3A%21missing%20status%3Apass%7Cstatus%3Aok%29&run_id=5147428859150336&run_id=6241478899138560&run_id=5199696832757760&run_id=5142037970550784&run_id=5093854074634240)

jgraham reviewed (approved):
If you're happy with the results, I'm happy with the `s/12/13/` in the config file :)

gsnedders committed:
Change the macOS display color profile on Azure Pipelines

This is effectively a no-op on the existing macOS 12 Azure Pipelines
images, as these are already configured to use sRGB. However, the
macOS 13 Azure Pipelines images are not, thus to avoid introducing
extra changes beyond those caused by the OS update we manually
configure this.

gsnedders commented on tools/ci/macos_color_profile.py:
```diff
@@ -0,0 +1,41 @@
+)
+
+
+def set_all_displays(profile_url):
```
c.f. https://github.com/WebKit/WebKit/blob/844e05173dcb29fdbddc42c3e843d28962cba7a6/Tools/DumpRenderTree/mac/LayoutTestHelper.m

gsnedders commented:
More [recent STP changes](https://wpt.fyi/results/?q=seq%28status%3Apass%7Cstatus%3Aok%20status%3Apass%7Cstatus%3Aok%20status%3Apass%7Cstatus%3Aok%20status%3Apass%7Cstatus%3Aok%20status%3A%21pass%26status%3A%21ok%26status%3A%21missing%29%20or%20seq%28status%3A%21pass%26status%3A%21ok%26status%3A%21missing%20status%3A%21pass%26status%3A%21ok%26status%3A%21missing%20status%3A%21pass%26status%3A%21ok%26status%3A%21missing%20status%3A%21pass%26status%3A%21ok%26status%3A%21missing%20status%3Apass%7Cstatus%3Aok%29&run_id=5190084075192320&run_id=6237988852334592&run_id=5177036501417984&run_id=5206136616124416&run_id=5129751730061312), with the same colour profile as the macOS 12 images.

My main Q here is whether we should just land this PR as is, or whether we want any other process for the colour profile change.

gsnedders commented:
> My main Q here is whether we should just land this PR as is, or whether we want any other process for the colour profile change.

Having discussed this in Matrix with @jgraham and @Ms2ger, we should just land this (because this makes it match the prior Azure Pipelines images), even if we have a broader discussion in RFC-land about whether we should require things about colour profiles.

gsnedders commented:
Curiously the infrastructure job is reproducibly failing with `TIMEOUT /infrastructure/testdriver/file_upload.sub.html`, whereas the full run did not.

gsnedders committed:
Make installing requirements to load all commands fallible

This is might happen if a command has requirements that are not
installable cross-platform, because the command is only intended to be
run on a single platform.

gsnedders commented on tools/wpt/wpt.py:
```diff
@@ -132,7 +132,11 @@ def create_complete_parser():
         props = commands[command]
 
-        venv.install_requirements(*props.get("requirements", []))
+        try:
+            venv.install_requirements(*props.get("requirements", []))
+        except Exception:
+            # this may happen if, e.g., a package is only installable on certain platforms
+            continue
```
cc @jgraham @Ms2ger 

I'm not sure this is necessarily something that makes sense; I could make all the requirements in `requirements_macos_color_profile.txt` `sys_platform == 'darwin'`, but then we just hit the same problem at import time just below.

Unless we go through and make all the PyObjC imports lazy behind a check that we're running on macOS?

jgraham replied:
Letting this fail would probably be fine. The more complete solution would be allowing the command definition to specify that the command itself should fail if you aren't on macOS.

jgraham replied:
Oh, sorry I missed that this was `create_complete_parser`. I think all it means is we don't generate documentation for this one command, right? So it's not great, but not totally awful either. Should probably still log a warning though.

jgraham reviewed (approved):
Well I can't really review the Cocoa-specific code, but otherwise this looks reasonable.
